docs/architecture/overview.md

```diff
@@ -47,8 +49,24 @@ The check before a transaction can e.g. programmatically check all of the parame
 > ⚠️ IMPORTANT: Since a guard has full power to block Safe transaction execution,
 > a broken guard can cause a denial of service for the Safe. Make sure to carefully audit the guard code and design recovery mechanisms.
 
+To learn more about guards, please refer to the [guards](./guards.md) documentation.
+
+#### Libraries
+
+Libraries implement custom execution flows that are optimized towards specific protocols or use cases. To use the libraries they are called using `delegatecall`, which executes the library logic via the Safe contract.
+
+> ⚠️ IMPORTANT: Since a libraries are called via delegate call, they can perform arbitrary state changes on the Safe contract. Make sure to carefully audit the libraries that are used.
+
+To learn more about libraries, please refer to the [libraries](./libraries.md) documentation.
+
 ### Technical Overview
 
+#### Safe Deployment
+
+The Safe contracts make heavy use of deterministic and counterfactual deployment. This not only makes it possible to reduce the gas costs but also guarantees that the deployed contracts cannot be arbitrarily changed. 
```

Comment:
I thought the benefit of counterfactual deployment is the deterministic address. Whether or not you use counterfactual deployment smart contracts are immutable so they can't be arbitrarily changed.

```suggestion
The Safe contracts make heavy use of counterfactual deployment. This makes it possible to reduce the gas costs and guarantees that the deployed contracts have deterministic addresses. 
```